include/envoy/upstream/upstream.h

```diff
@@ -261,16 +280,20 @@ class HostSet {
    *
    * @param hosts supplies the (usually new) list of hosts in the host set.
    * @param healthy hosts supplies the subset of hosts which are healthy.
+   * @param healthy hosts supplies the subset of hosts which are degraded and healthy.
```

Comment:
The number of params here and in all the places where we pass all of this stuff around is getting a little out of control IMO. Perhaps we should convert into a struct that we pass instead? It might be easier to document, have default empty, etc.? Thoughts?

Reply:
Yeah I want to slim down this call for sure, I'll see what using a param object looks like.